### PR TITLE
Code Review #2

### PR DESCRIPTION
**Pull request by hoguer**

### CODE REVIEW

File: db/init_db.js [hunk 1]
```diff
@@ -1,437 +1,443 @@
...
-} = require('./client');
+  getProductById, 
+  getAllProducts,
+  createProduct,
+} = require("./products")
+
 
 async function buildTables() {
   try {
+    console.log("HALP")
```
Comment (hoguer) [inline db/init_db.js]: let's remove this console.log

File: db/client.js [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 // database methods
 
 // export
-module.exports = {
-  client,
-  // db methods
-}
\ No newline at end of file
+module.exports = client
+  // db methods
```
Comment (hoguer) [inline db/client.js:10]: we can probably get rid of this comment and the `//export` comment

File: db/init_db.js [hunk 3]
```diff
@@ -1,437 +1,443 @@
...
   }
 }
 /* 
-
 Seed data 
-
 */
-
-
+console.log("Hellooooo!")
```
Comment (hoguer) [inline db/init_db.js]: let's get rid of this one, too

File: db/init_db.js [hunk 4]
```diff
@@ -1,437 +1,443 @@
...
 async function populateInitialData() {
+  console.log("Hellooooo!")
```
Comment (hoguer) [inline db/init_db.js]: this one, too

File: db/init_db.js [hunk 5]
```diff
@@ -1,437 +1,443 @@
...
   try {
     // create useful starting data
-    console.log("populating our wine and cheese tables")
-
+    console.log("populating our wine and cheese tables");
+    // const [name, description, imgURL, inStock, price, category] = await getAllProducts();
```
Comment (hoguer) [inline db/init_db.js]: can get rid of this comment

File: index.js [hunk 6]
```diff
@@ -22,7 +22,8 @@ server.use((req, res, next) => {
 });
 
 // bring in the DB connection
-const { client } = require('./db');
+const { client } = require('./db/client');
```
Comment (hoguer) [inline index.js]: we should remove this line

File: package.json [hunk 7]
```diff
@@ -18,11 +20,15 @@
...
     "bcrypt": "^5.0.1",
     "express": "^4.17.1",
+    "jsonwebtoken": "^8.5.1",
     "morgan": "^1.10.0",
     "pg": "^8.2.1",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-router-dom": "^5.2.0",
+    "react-router-dom": "^5.3.0",
+    "react": "^16.14.0",
```
Comment (hoguer) [inline package.json]: react, react-router-dom, and react-dom are repeated

File: routes/index.js [hunk 8]
```diff
@@ -1,11 +1,12 @@
 const apiRouter = require('express').Router();
-
 const productsRouter = require("./products");
 apiRouter.use("/product", productsRouter)
 
-const usersRouter = require("./users");
-apiRouter.use("/users", usersRouter)
 
+const productsRouter = require("./products");
```
Comment (hoguer) [inline routes/index.js:3]: this is duplicated from above.

File: routes/products.js [hunk 9]
```diff
@@ -1,24 +1,32 @@
 const productsRouter = require("express").Router();
 const { getProductById, getAllProducts, createProduct, getProductByName } = require("../db")
-
+const {
```
Comment (hoguer) [inline routes/products.js]: this is a duplicate of the line above it

File: routes/products.js [hunk 10]
```diff
@@ -1,24 +1,32 @@
...
 const { getProductById, getAllProducts, createProduct, getProductByName } = require("../db")
-
+const { 
+    getProductById, 
+    getAllProducts, 
+    createProduct, 
+    getProductByName } = require("../db")
+//get all products   
 productsRouter.get("/", async (req, res, next) =>{
+    console.log("Got here!")
```
Comment (hoguer) [inline routes/products.js]: if you are confident this is working, could get rid of this log

File: routes/products.js [hunk 11]
```diff
@@ -1,24 +1,32 @@
...
         res.send(allProducts)
     } catch (error) {
         throw error
     }
 } );
-
 productsRouter.get("/:productId", async (req, res, next) => {
+
+//get product by id 
+productsRouter.get("/:id", async (req, res, next) => {
```
Comment (hoguer) [inline routes/products.js:14]: this is a duplicate of the line above

File: src/components/App.js [hunk 12]
```diff
@@ -1,28 +1,81 @@
 import React, { useState, useEffect } from 'react';
-
+import ReactDOM from "react-dom";
```
Comment (hoguer) [inline src/components/App.js]: I believe this import is unused and can be removed

File: src/components/App.js [hunk 13]
```diff
@@ -1,28 +1,81 @@
...
+import { Routes, Route, NavLink, BrowserRouter as Router } from "react-router-dom";
```
Comment (hoguer) [inline src/components/App.js]: we don't need to import `BrowserRouter as Router`

File: src/components/App.js [hunk 14]
```diff
@@ -1,28 +1,81 @@
...
+//   // getSomething
+// } from '../api';
+
+import {
+  About,
+  Cart,
+  Home,
+  Login,
+  Products
+} from "../components"
```
Comment (hoguer) [inline src/components/App.js]: `from "."`

File: src/components/App.js [hunk 15]
```diff
@@ -1,28 +1,81 @@
...
+  //       setMessage(error.message);
+  //     });
+  // });
 
-  return (
+  return <> 
     <div className="App">
-      <h1>Hello, World!</h1>
-      <h2>{ message }</h2>
+      <h1>porto3000</h1>
```
Comment (hoguer) [inline src/components/App.js]: probably only want one `h1`

File: src/components/App.js [hunk 16]
```diff
@@ -1,28 +1,81 @@
...
+      <h1>Porto 3000</h1>
+      <nav className="navigation">
+        <div className="nav-links">
+          <NavLink to="/"> Home </NavLink> 
+          <NavLink to="/products"> Products </NavLink> 
+          <NavLink to="/about"> About Us</NavLink> 
+          <NavLink to="/checkout"> Checkout </NavLink>
```
Comment (hoguer) [inline src/components/App.js:45]: ok for class purposed, but IRL wouldn't want to leave a broken link in our code

File: src/components/App.js [hunk 17]
```diff
@@ -1,28 +1,81 @@
...
+              </>
+              :
+              <>
+                <NavLink to="/login"> Login </NavLink> 
+                <NavLink to="/register"> Register </NavLink> 
+              </>
+          }
+        </div>
+      </nav>
+      {/* <Routes>
```
Comment (hoguer) [inline src/components/App.js]: can get rid of this comment

File: src/components/Products.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+            console.log(response)
+            const result = await response.json();
+            console.log(result)
+            setProducts(result);
+        } catch (error) {
+            console.log("Trouble gathering products!", error)
+        }
+    }
+
+    // useEffect(fetchProducts, []);
```
Comment (hoguer) [inline src/components/Products.js]: sounds like this is in a PR, but let's use this line instead of calling it repeatedly.

File: index.js (hunk 6 above)
Comment (kbonifacio) [reply index.js]: complete

File: package.json (hunk 7 above)
Comment (kbonifacio) [reply package.json]: complete

File: routes/products.js (hunk 10 above)
Comment (kbonifacio) [reply routes/products.js]: complete

File: routes/products.js (hunk 11 above)
Comment (kbonifacio) [reply routes/products.js:14]: complete

File: src/components/App.js (hunk 12 above)
Comment (kbonifacio) [reply src/components/App.js]: complete

File: src/components/App.js (hunk 13 above)
Comment (kbonifacio) [reply src/components/App.js]: complete

File: src/components/App.js (hunk 14 above)
Comment (kbonifacio) [reply src/components/App.js]: complete

File: src/components/App.js (hunk 15 above)
Comment (kbonifacio) [reply src/components/App.js]: complete

File: src/components/App.js (hunk 17 above)
Comment (kbonifacio) [reply src/components/App.js]: complete

File: src/components/Products.js (hunk 18 above)
Comment (kbonifacio) [reply src/components/Products.js]: complete

**Review by kbonifacio — APPROVED** [review]
Lgtm

**Pull request merged**